### ISSUE TITLE
test/format sanitizer assertion failure

### ISSUE THREAD

**Issue by sueloverso** [wiredtiger/wiredtiger#1779]
Two successive failures in test/format relating to locking hit the same assertion.

build.wiredtiger.com:8080/job/wiredtiger-test-format-stress-sanitizer/2975
build.wiredtiger.com:8080/job/wiredtiger-test-format-stress-sanitizer/2976

```
[1426697115:464368][89103:0x7f191b7fe700], t, file:wt, statlog-server: ../src/session/session_dhandle.c, 408: F_ISSET(session, WT_SESSION_SCHEMA_LOCKED) || !F_ISSET(session, WT_SESSION_HANDLE_LIST_LOCKED | WT_SESSION_NO_SCHEMA_LOCK | WT_SESSION_TABLE_LOCKED)
[1426697115:464419][89103:0x7f191b7fe700], t, file:wt, statlog-server: aborting WiredTiger library
```
One CONFIG is:
```
############################################
#  RUN PARAMETERS
############################################
abort=0
auto_throttle=1
firstfit=0
bitcnt=7
bloom=1
bloom_bit_count=38
bloom_hash_count=22
bloom_oldest=0
cache=73
checkpoints=1
checksum=uncompressed
chunk_size=5
compaction=0
compression=bzip
data_extend=0
data_source=file
delete_pct=36
dictionary=0
evict_max=2
file_type=row-store
backups=0
huffman_key=0
huffman_value=0
insert_pct=24
internal_key_truncation=1
internal_page_max=13
isolation=read-uncommitted
key_gap=9
key_max=49
key_min=25
leak_memory=0
leaf_page_max=12
logging=1
logging_archive=0
logging_prealloc=1
lsm_worker_threads=4
merge_max=14
mmap=1
ops=100000
prefix_compression=1
prefix_compression_min=6
repeat_data_pct=69
reverse=0
rows=100000
runs=1
split_pct=67
statistics=0
statistics_server=1
threads=16
timer=20
value_max=3654
value_min=17
wiredtiger_config=
write_pct=89
############################################
```

**Comment by sueloverso** [comment]
Since it was sanitizer running, the assertion failure did not generate a core file.  I am running the config to see how readily reproducible it is.

**Comment by sueloverso** [comment]
It hit on the 2nd iteration of the above CONFIG file.  Here is a stack:
```
#4  0x000000000045ecaf in __wt_assert (session=0x899ff0, error=0, 
    file_name=0x4fd558 "../src/session/session_dhandle.c", line_number=408, 
    fmt=0x4fd555 "%s") at ../src/support/err.c:464
#5  0x000000000045d5ec in __wt_session_get_btree (session=0x899ff0, 
    uri=0x8d0bb0 "file:wt", checkpoint=0x0, cfg=0x0, flags=8)
    at ../src/session/session_dhandle.c:406
#6  0x00000000004a1622 in __conn_btree_apply_internal (session=0x899ff0, 
    dhandle=0xa024a0, func=0x4154e7 <__statlog_apply>, cfg=0x0)
    at ../src/conn/conn_dhandle.c:484
#7  0x00000000004a192c in __wt_conn_btree_apply (session=0x899ff0, 
    apply_checkpoints=0, uri=0x0, func=0x4154e7 <__statlog_apply>, cfg=0x0)
    at ../src/conn/conn_dhandle.c:536
#8  0x0000000000415b58 in __statlog_log_one (session=0x899ff0, 
    path=0x7f429abecf10, tmp=0x7f429abecee0) at ../src/conn/conn_stat.c:331
#9  0x0000000000415e3d in __statlog_server (arg=0x899ff0)
    at ../src/conn/conn_stat.c:418
```
@agorrod I'm assigning to you as it is probably related to 88066d7b504300954.

**Comment by sueloverso** [comment]
If I revert that one change this CONFIG runs just fine (> 12 iterations now).

**Comment by agorrod** [comment]
Fixed by #1780 - closing.
